lib/logstash/outputs/scalyr.rb

```diff
@@ -676,49 +689,51 @@ def get_stats
   # uploading to Scalyr over time.
   def send_status
 
-    status_event = {
-      :ts => (Time.now.to_f * (10**9)).round,
-      :attrs => {
-        'logfile' => "scalyr_logstash.log",
-        'plugin_id' => self.id,
+    @send_stats.synchronize do
```

Comment:
Because of the indentation, I could have misread that, but it looks like everything (including sending the request) is under ``send_stats`` lock, right? If indeed the case, we should move sending of the request out of it since that could take a while and should already be thread-safe.